File: binarygap/c/binarygap.c
#include <stdio.h>
#include "binarygap.h"

int solution(int N)
{
    int startgap = 0;
    int max = 0, gap = 0, modN;

    while (N != 0)
    {
        modN = N % 2;
        if (modN == 1) {
            if (gap > max)
                max = gap;
            startgap = 1;
            gap = 0;
        } else {
            if (startgap == 1) {
                gap += 1;
            }
        }
        N = (int) (N / 2);
        // printf("N: %d modN: %d\n", N, modN);
    }

    return max;
}

File: 01-iterations/binary-gap/python/binary_gap.py

def solution(N):
    strN = str(bin(N))

    max = gap = 0
    for char in strN[2:]:
        if char == "1":
            if gap > max:
                max = gap
            gap = 0
        else:
            gap += 1

    return max


if __name__ == "__main__":
    print(solution(32))
